ports/espressif/common-hal/wifi/__init__.c

```diff
@@ -199,11 +199,20 @@ void common_hal_wifi_init(bool user_initiated) {
         ESP_LOGE(TAG, "WiFi error code: %x", result);
         return;
     }
-    // set the default lwip_local_hostname
-    char cpy_default_hostname[strlen(CIRCUITPY_BOARD_ID) + (MAC_ADDRESS_LENGTH * 2) + 6];
+    // Set the default lwip_local_hostname capped at 32 characters. We trim off
+    // the start of the board name (likely manufacturer) because the end is
+    // often more unique to the board.
+    size_t board_len = MIN(32 - ((MAC_ADDRESS_LENGTH * 2) + 6), strlen(CIRCUITPY_BOARD_ID));
+    size_t board_trim = strlen(CIRCUITPY_BOARD_ID) - board_len;
+    // Avoid double _ in the hostname.
+    if (CIRCUITPY_BOARD_ID[board_trim] == '_') {
+        board_trim++;
+    }
+
+    char cpy_default_hostname[board_len + (MAC_ADDRESS_LENGTH * 2) + 6];
     uint8_t mac[MAC_ADDRESS_LENGTH];
     esp_wifi_get_mac(ESP_IF_WIFI_STA, mac);
-    sprintf(cpy_default_hostname, "cpy_%s_%x", CIRCUITPY_BOARD_ID, (unsigned int)mac);
+    snprintf(cpy_default_hostname, sizeof(cpy_default_hostname), "cpy-%s-%x", CIRCUITPY_BOARD_ID + board_trim, (unsigned int)mac);
```

Comment:
My original code to convert the MAC address doesn't work, we should use something closer to your original code in server.c:
```py
snprintf(cpy_default_hostname, sizeof(cpy_default_hostname), "cpy-%s-%02x%02x%02x%02x%02x%02x", CIRCUITPY_BOARD_ID + board_trim, mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
```

Reply:
I wondered where that `t` came from... it's the end of `Adafruit_...`. Would be nice to split it at a `_`, but probably not worth the code.

Reply:
I looked at going to the next _ but in some cases you do lose something helpful. In some cases the remainder is the chip number (such as 6 from esp32c6) for example.

Reply:
could drop to 3 octets of MAC, to free up more of the board name

Reply:
I looked what dropping one octet would do and it didn't seem too helpful. I thought about xoring the three octets together to ensure all bits are included.